### PR TITLE
Packages for OCaml 5.1.0

### PR DESCRIPTION
**Pull request by Octachron**
The new packages for the OCaml 5.1.0 release:

- ocaml-system.5.1.0
- ocaml-base-compiler.5.1.0
- ocaml-variants.5.1.0+options
- ocaml-src.5.10
- ocaml (for 5.1.1
- ocaml-variants.5.1.1+trunk

I will send the ocaml-manual package in a separate PR.

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
Thanks!

**Comment by MisterDA** [comment]
Would it make sense for the OCaml packages to require libzstd as a depext, maybe optionally? Support for marshalling data using zstd was added in 5.1.

File: packages/ocaml-variants/ocaml-variants.5.1.0+options/opam [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+post-messages: [
+  "A failure in the middle of the build may be caused by build parallelism
+   (enabled by default).
+   Please file a bug report at https://github.com/ocaml/opam-repository/issues"
+  {failure & jobs > 1}
+  "You can try installing again including --jobs=1
+   to force a sequential build instead."
+  {failure & jobs > 1 & opam-version >= "2.0.5"}
+]
+conflicts: [ "ocaml-option-fp" ]
```
Comment (copy) [inline packages/ocaml-variants/ocaml-variants.5.1.0+options/opam:74]: Is this correct? I believe frame pointers have been restored in 5.1, and it wasn't present in 5.1.0~rc3+options.

Comment (Octachron) [reply packages/ocaml-variants/ocaml-variants.5.1.0+options/opam:74]: This is a mistake indeed, thanks a lot for spotting it !

**Pull request merged**